From 84769953fda853172a312980f07b0d9a550402ce Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Wed, 24 Jun 2020 12:41:27 +0800
Subject: [PATCH] clk: imx: imx8mm: solve the system hang when starting

solve the system hang when starting. Set the clock before accessing
the OCOTP register, otherwise it will cause the system to hang

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/clk/imx/clk-imx8mm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/imx/clk-imx8mm.c b/drivers/clk/imx/clk-imx8mm.c
index 35d4df63d47b..137f8352e8c9 100644
--- a/drivers/clk/imx/clk-imx8mm.c
+++ b/drivers/clk/imx/clk-imx8mm.c
@@ -401,6 +401,7 @@ static int const clks_init_on[] __initconst = {
 	IMX8MM_CLK_DRAM_APB, IMX8MM_CLK_A53_DIV,
 	IMX8MM_ARM_PLL_OUT, IMX8MM_CLK_DISP_AXI,
 	IMX8MM_CLK_DISP_APB,
+	IMX8MM_CLK_OCOTP_ROOT,
 };
 
 static struct clk ** const uart_clks[] __initconst = {
-- 
2.17.1

